crypto/evp/keymgmt_meth.c

```diff
@@ -0,0 +1,197 @@
+                break;
+            keymgmt->importdomain_types =
+                OSSL_get_OP_keymgmt_importdomain_types(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_EXPORTDOMAIN_TYPES:
+            if (keymgmt->exportdomain_types != NULL)
+                break;
+            keymgmt->exportdomain_types =
+                OSSL_get_OP_keymgmt_exportdomain_types(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_IMPORTKEY_PRIV:
+            if (keymgmt->importkey_priv != NULL)
+                break;
+            keymgmt->importkey_priv =
+                OSSL_get_OP_keymgmt_importkey_priv(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_IMPORTKEY_PUB:
+            if (keymgmt->importkey_pub != NULL)
+                break;
+            keymgmt->importkey_pub = OSSL_get_OP_keymgmt_importkey_pub(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_GENKEY:
+            if (keymgmt->genkey != NULL)
+                break;
+            keymgmt->genkey = OSSL_get_OP_keymgmt_genkey(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_LOADKEY:
+            if (keymgmt->loadkey != NULL)
+                break;
+            keymgmt->loadkey = OSSL_get_OP_keymgmt_loadkey(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_FREEKEY:
+            if (keymgmt->freekey != NULL)
+                break;
+            keymgmt->freekey = OSSL_get_OP_keymgmt_freekey(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_EXPORTKEY_PRIV:
+            if (keymgmt->exportkey_priv != NULL)
+                break;
+            keymgmt->exportkey_priv =
+                OSSL_get_OP_keymgmt_exportkey_priv(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_EXPORTKEY_PUB:
+            if (keymgmt->exportkey_pub != NULL)
+                break;
+            keymgmt->exportkey_pub = OSSL_get_OP_keymgmt_exportkey_pub(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_IMPORTKEY_PRIV_TYPES:
+            if (keymgmt->importkey_priv_types != NULL)
+                break;
+            keymgmt->importkey_priv_types =
+                OSSL_get_OP_keymgmt_importkey_priv_types(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_IMPORTKEY_PUB_TYPES:
+            if (keymgmt->importkey_pub_types != NULL)
+                break;
+            keymgmt->importkey_pub_types =
+                OSSL_get_OP_keymgmt_importkey_pub_types(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_EXPORTKEY_PRIV_TYPES:
+            if (keymgmt->exportkey_priv_types != NULL)
+                break;
+            keymgmt->exportkey_priv_types =
+                OSSL_get_OP_keymgmt_exportkey_priv_types(fns);
+            break;
+        case OSSL_FUNC_KEYMGMT_EXPORTKEY_PUB_TYPES:
+            if (keymgmt->exportkey_pub_types != NULL)
+                break;
+            keymgmt->exportkey_pub_types =
+                OSSL_get_OP_keymgmt_exportkey_pub_types(fns);
+            break;
+        }
+    }
+    keymgmt->prov = prov;
```

Comment:
Should we have some consistency checking here? For example does it ever make sense to not have a freekey function when you have a loadkey/genkey function? Or to not have importkey_pub, when importkey_pub_types is non NULL? etc

Reply:
Good point.  Will add a check

Reply:
Done